simpa/core/simulation_modules/reconstruction_module/reconstruction_module_delay_and_sum_adapter.py

```diff
@@ -56,36 +56,33 @@ def reconstruction_algorithm(self, time_series_sensor_data, detection_geometry:
                                       detection_geometry: DetectionGeometryBase,
-                                      settings: dict = None,
                                       speed_of_sound_in_m_per_s: int = 1540,
                                       time_spacing_in_s: float = 2.5e-8,
-                                      sensor_spacing_in_mm: float = 0.1) -> np.ndarray:
+                                      sensor_spacing_in_mm: float = 0.1,
+                                      recon_mode: str = Tags.RECONSTRUCTION_MODE_PRESSURE,
+                                      apodization: str = Tags.RECONSTRUCTION_APODIZATION_BOX) -> np.ndarray:
     """
     Convenience function for reconstructing time series data using Delay and Sum algorithm implemented in PyTorch
 
     :param time_series_sensor_data: (2D numpy array) sensor data of shape (sensor elements, time steps)
     :param detection_geometry: The DetectionGeometryBase that should be used to reconstruct the given time series data
-    :param settings: (dict) settings dictionary: by default there is none and the other parameters are used instead,
-                     but if parameters are given in the settings those will be used instead of parsed arguments)
     :param speed_of_sound_in_m_per_s: (int) speed of sound in medium in meters per second (default: 1540 m/s)
     :param time_spacing_in_s: (float) time between sampling points in seconds (default: 2.5e-8 s which is equal to 40 MHz)
     :param sensor_spacing_in_mm: (float) space between sensor elements in millimeters (default: 0.1 mm)
+    :param recon_mode: SIMPA Tag defining the reconstruction mode - pressure default OR differential
+    :param apodization: SIMPA Tag defining the apodization function (default box)
     :return: (2D numpy array) reconstructed image as 2D numpy array
     """
 
-    # create settings if they don't exist yet
-    if settings is None:
-        settings = Settings()
-
-    # parse reconstruction settings if they are not given in the settings
-    if Tags.DATA_FIELD_SPEED_OF_SOUND not in settings or settings[Tags.DATA_FIELD_SPEED_OF_SOUND] is None:
-        settings[Tags.DATA_FIELD_SPEED_OF_SOUND] = speed_of_sound_in_m_per_s
-
-    if Tags.SENSOR_SAMPLING_RATE_MHZ not in settings or settings[Tags.SENSOR_SAMPLING_RATE_MHZ] is None:
-        settings[Tags.SENSOR_SAMPLING_RATE_MHZ] = (1.0 / time_spacing_in_s) / 1000000
-
-    if Tags.SPACING_MM not in settings or settings[Tags.SPACING_MM] is None:
-        settings[Tags.SPACING_MM] = sensor_spacing_in_mm
+    # create settings
+    settings = Settings()
+    settings.set_reconstruction_settings({
+        Tags.DATA_FIELD_SPEED_OF_SOUND: speed_of_sound_in_m_per_s,
+        Tags.SPACING_MM: sensor_spacing_in_mm,
+        Tags.RECONSTRUCTION_APODIZATION_METHOD: apodization,
+        Tags.RECONSTRUCTION_MODE: recon_mode,
+        Tags.SENSOR_SAMPLING_RATE_MHZ: (1.0 / time_spacing_in_s) / 1000000
+    })
```

Comment:
To avoid the duplication of these lines in DAS, DMAS and sDMAS adapters you could create a function, for example in `reconstruction_module/__init__.py` in the `ReconstructionAdapterBase` (perhaps there even is a better location) which generates the reconstruction settings based on parameters:

```python
def create_reconstruction_settings(self, speed_of_sound_in_m_per_s: int = 1540,
                                    time_spacing_in_s: float = 2.5e-8,
                                    sensor_spacing_in_mm: float = 0.1,
                                    recon_mode: str = Tags.RECONSTRUCTION_MODE_PRESSURE,
                                    apodization: str = Tags.RECONSTRUCTION_APODIZATION_BOX) -> Settings:
       '''
       TODO: Add description of function including parameters and return value
       '''
        # create settings
        settings = Settings()
        settings.set_reconstruction_settings({
            Tags.DATA_FIELD_SPEED_OF_SOUND: speed_of_sound_in_m_per_s,
            Tags.SPACING_MM: sensor_spacing_in_mm,
            Tags.RECONSTRUCTION_APODIZATION_METHOD: apodization,
            Tags.RECONSTRUCTION_MODE: recon_mode,
            Tags.SENSOR_SAMPLING_RATE_MHZ: (1.0 / time_spacing_in_s) / 1000000
        })
        return settings
```

Then the convenience functions can simply be turned into something like this:
```python
settings = ReconstructionAdapterBase.create_reconstruction_settings(speed_of_sound_in_m_per_s, time_spacing_in_s, sensor_spacing_in_mm, recon_mode, apodization)
    
    adapter = DelayAndSumAdapter(settings)
    return adapter.reconstruction_algorithm(time_series_sensor_data, detection_geometry)
```